gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/workspace/HasBundlesDirListener.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.gradle.core.workspace;
+
+import org.eclipse.sapphire.FilteredListener;
+import org.eclipse.sapphire.PropertyContentEvent;
+
+/**
+ * @author Andy Wu
+ */
+public class HasBundlesDirListener extends FilteredListener<PropertyContentEvent>
```

Comment:
Hey @AndyWu2015 this doesn't logically make sense.  HasBundlesDir is a derived/readonly property (it doesn't change because a location either "has" or "doesn't have" a bundles dir.  It will be changing.

Why do you need provisionLiferayBundle to change from true/false?  shouldn't only the user be making that change?